ts/screens/authentication/cie/CieConsentDataUsageScreen.tsx

```diff
@@ -2,230 +2,172 @@
+    if (hasError && errorCode === "22") {
+      trackLoginCieDataSharingError();
     }
-    if (this.state.hasError) {
-      if (this.state.errorCode === "22") {
-        trackLoginCieDataSharingError();
+  }, [errorCode, hasError]);
+
+  useEffect(() => {
+    if (hasError) {
+      if (errorCode === "1002") {
+        navigation.navigate(ROUTES.AUTHENTICATION, {
+          screen: ROUTES.UNLOCK_ACCESS_SCREEN,
+          params: { authLevel: "L2" }
+        });
+        return;
       }
-      if (this.state.errorCode === "1002") {
-        return <UnlockAccessScreen identifier="CIE" />;
-      } else {
-        const errorTranslationKey = this.state.errorCode
-          ? `authentication.errors.spid.error_${this.state.errorCode}`
-          : "authentication.errors.network.title";
-        return (
-          <GenericErrorComponent
-            retryButtonTitle={I18n.t(
-              "authentication.cie.dataUsageConsent.retryCTA"
-            )}
-            onRetry={this.props.resetNavigation}
-            onCancel={undefined}
-            image={require("../../../../img/broken-link.png")} // TODO: use custom or generic image?
-            text={I18n.t(errorTranslationKey, {
-              defaultValue: I18n.t("authentication.errors.spid.unknown")
-            })}
-          />
-        );
-      }
-    } else {
-      return (
-        <WebView
-          androidCameraAccessDisabled={true}
-          androidMicrophoneAccessDisabled={true}
-          textZoom={100}
-          originWhitelist={originSchemasWhiteList}
-          source={{ uri: decodeURIComponent(this.cieAuthorizationUri) }}
-          javaScriptEnabled={true}
-          onShouldStartLoadWithRequest={this.handleShouldStartLoading}
-          renderLoading={() => loaderComponent}
-          onError={this.handleWebViewError}
-          onHttpError={this.handleHttpError}
-        />
-      );
+      navigation.navigate(ROUTES.AUTHENTICATION, {
+        screen: ROUTES.AUTH_ERROR_SCREEN,
+        params: { errorCode }
+      });
     }
-  };
+  }, [errorCode, hasError, navigation]);
 
-  public render(): React.ReactNode {
-    const goBack = this.state.hasError ? false : this.showAbortAlert;
+  if (isLoginSuccess) {
+    return <LoaderComponent />;
+  }
+  if (!hasError) {
     return (
-      <TopScreenComponent
-        goBack={goBack}
-        headerTitle={withTrailingPoliceCarLightEmojii(
-          I18n.t("authentication.cie.genericTitle"),
-          this.props.isCieUatEnabled
-        )}
-      >
-        {this.getContent()}
-      </TopScreenComponent>
+      <WebView
+        testID="webview-cie-test"
+        androidCameraAccessDisabled={true}
+        androidMicrophoneAccessDisabled={true}
+        textZoom={100}
+        originWhitelist={originSchemasWhiteList}
+        source={{ uri: decodeURIComponent(cieConsentUri) }}
+        javaScriptEnabled={true}
+        onShouldStartLoadWithRequest={handleShouldStartLoading}
+        renderLoading={() => <LoaderComponent />}
```

Comment:
```suggestion
        renderLoading={LoaderComponent}
```

Reply:
this modification cannot be made because the webview field renderLoading expects a function that returns a loader, in fact if I try to apply your suggestion, as I already did, I get this error:
`Type 'Element' is not assignable to type '(() => ReactElement<any, string | JSXElementConstructor<any>>) & Element'`